### ISSUE TITLE
Allow color settings to save to Profile, instead of workspace.

### ISSUE THREAD

**Issue by AferraroMezrah** [johnpapa/vscode-peacock#517]
Now that Profiles have left beta, I think it's reasonable to allow peacock settings to be saved based on profile and not just workspace.

I think it would be best to have peacock save the color setting to the active profile that is running. 

Given the original purpose of peacock is project based colors I can see why you might want to leave it alone. But I think profiles mirror the workspace spirit in that it can be different for any number of projects.

For us, when we share workspace files to make sure the entire team is using the same settings for certain connections/extensions, not everyone wants a color change.

**Comment by johnpapa** [comment]
Thank you @AferraroMezrah for creating this issue!

**Comment by robokozo** [comment]
Glad to see this has been requested! I'd hate to force my questionable color choices on others :)

**Comment by thoroc** [comment]
Would that be covered by https://github.com/johnpapa/vscode-peacock/issues/7 ?

**Comment by noisy** [comment]
I would love to see a progress on that

**Comment by jadnhm** [comment]
I think @thoroc is right this is a duplicate of https://github.com/johnpapa/vscode-peacock/issues/7

**Comment by thoroc** [comment]
Thanks for the ping. I completely forgot all about that one. :)

**Comment by AferraroMezrah** [comment]
@thoroc #7 was written before profiles though. I think it makes sense to include that information on #7, although activity has been slow on that request. Or keep the dedicated profiles request here. Either way!

**Comment by jadnhm** [comment]
@ncamaa I'm not sure I understand what you're suggesting. It's possible you don't understand the limitation described by https://github.com/johnpapa/vscode-peacock/issues/7#issuecomment-1921946816

If you want to see this issue resolved please pester the VSCode team about this issue
https://github.com/microsoft/vscode/issues/40233

**Comment by AlexSapoznikov** [comment]
I rarely close my workspace. Could we just have an option to not save anything at all? It's easier to re-apply theme from favourites when opening workspace.

**Comment by jadnhm** [comment]
@AlexSapoznikov  I think it's possible you don't understand the limitation described by https://github.com/johnpapa/vscode-peacock/issues/7#issuecomment-1921946816

- tl;dr: there is no way to tell VSCode to change the colours WITHOUT SAVING TO THE SETTINGS FILE

If you want to see this issue resolved please pester the VSCode team about this issue
https://github.com/microsoft/vscode/issues/40233

**Comment by jadnhm** [comment]
@AlexSapoznikov 
>So you are saying that you can not apply new colors without touching .vscode/settings.json?
yes that's exactly what I'm saying. Please read the linked comment from #7 and peruse that issue's other comments for more specifics
https://github.com/johnpapa/vscode-peacock/issues/7#issuecomment-1921946816
<img width="773" alt="image" src="https://github.com/johnpapa/vscode-peacock/assets/26800648/962a3709-4f98-494a-951f-d96c5b18e5e6">


What Peacock is attempting is not 'changing the theme' although that's something that could be attempted (probably by another extension?). Peacocks job is to leave EVERY OTHER SETTING besides window 'chrome' colours _alone_, and themes consist of many settings besides colour (such as fonts, for example)

> I would just prefer using this without resetting git all the time
we all would prefer that which is why we're all so angry that https://github.com/microsoft/vscode/issues/40233 has been ignored for so long and why I directed you there in my previous comment.

**Comment by AlexSapoznikov** [comment]
@jadnhm I did read the whole thread before testing. I just managed to misunderstand that vscode settings is also needed for applying colors and not only for restoring. In that case we can close this thread.
Thanks for clarifying!

**Comment by AferraroMezrah** [comment]
> @AferraroMezrah Please read my summary of #7 here [#7 (comment)](https://github.com/johnpapa/vscode-peacock/issues/7#issuecomment-1921946816)
> 
> You can see that the 'location where the colour information is saved' is dependant upon VSCode itself and unless someone can figure out an amazing hack there is not much https://github.com/johnpapa/vscode-peacock/ can do about it, unfortunately

I understand the limitation of the requirement to write to the settings.json file now. But the original question was about if the "Profile" feature of vscode can be used here. Does the profile create it's own settings.json? How does VSCode treat profiles.

Right now on my team we share a workspace configuration, but I've noticed that I can use different profiles without affecting our source control. And that's what led to the question, to both better understand profiles and what they do under the hood, and also see if that allows them to be used to control peacock settings in a way that isn't the workspace. Because clearly profiles are storing settings configurations _per_ profile somehow. I was hoping it would apply here?

**Comment by sneridagh** [comment]
Is there any news about this one? Like @AferraroMezrah said, what if one can't modify the `.vscode/settings.json` file, because it's versioned, a thing that is quite common nowadays.

**Comment by johnpapa** [comment]
Closing — VS Code's extension API has no `ConfigurationTarget.Profile`. Extensions can only write to Global, Workspace, or WorkspaceFolder scopes. The best alternative is #495 (user-level workspace→color mappings) which would avoid writing personal colors to shared workspace files entirely. Follow #495 for progress. Thanks! 🙏
